code/game/objects/items/implants/implantswap.dm

```diff
@@ -0,0 +1,74 @@
+/*!
+The swapper implant comes in two parts, the activator pen, and the swapper implant itself. The activator stores the object reference to the implant, and uses that to swap the locations of the mobs holding the objects.area
+*/
+
+/obj/item/pen/swap_activator
+	///Stores the implant object ref.
+	var/obj/item/implant/swapper/receiver
+
+/obj/item/pen/swap_activator/proc/activate(var/obj/item/implant/swapper/R = receiver, var/obj/A = src)
+	///Grab the recievers mob
+	var/mob/living/RL = R.loc
+	///Grab the activators mob
+	var/mob/living/AL = A.loc
+	
+	if(R.implanted && R && A)
+		///Grab the recievers turf
+		var/turf/RT = get_turf(RL.loc)
+		///Grab the activators turf
+		var/turf/AT = get_turf(A.loc)
+		
+		playsound(RL, 'sound/effects/sparks4.ogg')
+		playsound(AL, 'sound/effects/sparks4.ogg')
+
+		RL.visible_message("[RL] suddenly vanishes, leaving [AL] in their place!")
```

Comment:
Visible messages can also send chatlogs to the user as a second argument, use that instead of just doing the to_chats below again.

Reply:
I honestly prefer to keep my functions atomic, which is why I use to_chat, also visible message doesn't support HTML spans I believe.

Reply:
visible_message allows for HTML, and these probably should be span'd with notice or alternative. In general I don't think unstyled text is wanted, and something I've been asked to change before

Reply:
still the point about atomic, but I will introduce the notices.